ui/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -199,6 +203,17 @@ export default class PermissionPageContainer extends Component {
     };
     ///: END:ONLY_INCLUDE_IF
 
+    const containsEthPermissionsAndNonEVMAccount = (accounts, permissions) => {
+      const containsEthPermissions = Object.keys(permissions).some(
+        (permission) => Object.keys(RestrictedEthMethods).includes(permission),
+      );
+      const containsNonEVMAccount = accounts.some(
+        (account) => !isEvmAccountType(account.type),
+      );
+
+      return containsEthPermissions && containsNonEVMAccount;
+    };
```

Comment:
This seems like an utility method, I don't see a reason for it to be inside a component unless I'm missing something. Also, unit tests.